test/org/sagebionetworks/bridge/services/ConsentServiceImplTest.java

```diff
@@ -179,162 +188,194 @@ public void cannotConsentIfTooYoung() {
+        status = ConsentStatus.forSubpopulation(consentService.getConsentStatuses(context), subpopulation.getGuid());
+        assertTrue("Should be consented", status.isConsented());
+        assertFalse("New consent activated. Should no longer have signed most recent consent. ", status.getSignedMostRecentConsent());
 
         // To consent again, first need to withdraw. User is consented and has now signed most recent consent.
-        consentService.withdrawConsent(testUser.getStudy(), testUser.getUser(), WITHDRAWAL, UNIX_TIMESTAMP);
-        consentService.consentToResearch(testUser.getStudy(), testUser.getUser(),
+        consentService.withdrawConsent(testUser.getStudy(), subpopulation.getGuid(), testUser.getUser(), WITHDRAWAL, UNIX_TIMESTAMP);
+        
+        consentService.consentToResearch(testUser.getStudy(), subpopulation.getGuid(), testUser.getUser(),
             new ConsentSignature.Builder().withConsentSignature(consent).withSignedOn(DateTime.now().getMillis()).build(),
             SharingScope.SPONSORS_AND_PARTNERS, false);
 
-        assertTrue("Should still be consented.",
-                consentService.hasUserConsentedToResearch(testUser.getStudy(), testUser.getUser()));
-        assertTrue("Should again have signed most recent consent.",
-                consentService.hasUserSignedActiveConsent(testUser.getStudy(), testUser.getUser()));
+        status = ConsentStatus.forSubpopulation(consentService.getConsentStatuses(context), subpopulation.getGuid());
+        assertTrue("Should still be consented.", status.isConsented());
+        assertTrue("Should again have signed most recent consent.", status.getSignedMostRecentConsent());
     }
     
     @Test
     public void withdrawConsent() {
         long originalSignedOn = DateTime.now().getMillis();
-        ConsentSignature consent = new ConsentSignature.Builder().withName("John Smith")
-                .withImageData("data").withImageMimeType("image/png").withBirthdate("1990-11-11")
-                .withSignedOn(originalSignedOn).build();
+        ConsentSignature consent = makeSignature(originalSignedOn);
 
-        consentService.consentToResearch(testUser.getStudy(), testUser.getUser(), consent,
+        consentService.consentToResearch(testUser.getStudy(), subpopulation.getGuid(), testUser.getUser(), consent,
                 SharingScope.SPONSORS_AND_PARTNERS, false);
 
         // Consent exists, user is consented
-        assertTrue(consentService.hasUserConsentedToResearch(testUser.getStudy(), testUser.getUser()));
-        assertTrue(testUser.getUser().doesConsent());
-        assertNotNull(consentService.getConsentSignature(testUser.getStudy(), testUser.getUser()));
+        ConsentStatus status = consentService.getConsentStatuses(context).get(0);
+        assertTrue(status.isConsented());
+        assertNotNull(consentService.getConsentSignature(testUser.getStudy(), subpopulation.getGuid(), testUser.getUser()));
```

Comment:
Rewrote so the status objects in and out of the user, as well as the boolean calculated from the collection, are all tested every time in helper assertion methods.